spec/features/guests/guest_visits_questions_index_spec.rb

```diff
@@ -20,4 +27,41 @@
       expect(page).to have_content(second_question.find_user)
     end
   end
+
+  it 'should have a list of categories' do
+    category_two    = Fabricate(:category, name:"Halumi")
+
+    visit questions_path
+
+    within ('.categories') do
+      expect(page).to have_link(category.name)
+      expect(page).to have_link(category_two.name)
+
+      expect(page).to_not have_link("Fordham")
+    end
+  end
+
+  it 'should sort questions by category' do
+    second_user     = Fabricate(:user, name:"Quade")
+    category_two    = Fabricate(:category, name:"Halumi")
+    third_question  = Fabricate(:question, user:second_user, category:category_two, title:"Rust Buckets?")
+
+    visit questions_path
+
+    within ('.categories') do
+      click_link(category.name)
+    end
+
+    expect(current_path).to eq(questions_path)
+
+    within('.questions-by-category') do
+      expect(page).to have_content(first_question.title)
+      expect(page).to have_content(first_question.find_user)
+      expect(page).to have_content(second_question.title)
```

Comment:
The assertions in your test are great. I really like that you're checking that some elements are present as well as checking that some are not.